payments_service/base/tests/__init__.py

```diff
@@ -107,6 +107,7 @@ def endpoint(self, view):
         dynamic_urls.urlpatterns = patterns(
             '',
             url(r'^dynamic-endpoint$', view),
+            url(r'^dynamic-endpoint/(?P<pk>[^/]+)?/?$', view),
```

Comment:
Isn't that covered by the previous line? This new line would match things like dynamic-endpoint//